### PR TITLE
Improved error handling & loading

### PR DESCRIPTION
**Pull request by bartolomej**
Wallet modal now shows a dismissable error state in a modal:

Screenshot 2022-10-08 at 10 37 50

How this looks in action (usage in [Flowser](https://github.com/onflowser/flowser)):

https://user-images.githubusercontent.com/36109955/194699797-0267d62d-ec32-45df-ab86-b7213695eedc.mov

### CODE REVIEW

File: src/comps/err.comp.tsx [hunk 1]
```diff
@@ -1,12 +1,28 @@
...
+
 export type StackError = {
   stack: string
 }
 
-export function Err({error}: {error: StackError}) {
-  if (error == null) return null
+type Error = StackError | string
+
+export function Err({error}: {error: Error}) {
```
Comment (huyndo) [inline src/comps/err.comp.tsx:11]:
```suggestion
export function Err({error, title="Error occurred"}: {error: Error, title?: string}) {
```

File: src/comps/err.comp.tsx [hunk 2]
```diff
@@ -1,12 +1,28 @@
...
   return (
-    <div>
-      <pre>{error.stack}</pre>
-    </div>
+    <Dialog>
+      <Themed.h1 sx={{textAlign: "center", mb: 20}}>Error occurred</Themed.h1>
```
Comment (huyndo) [inline src/comps/err.comp.tsx:14]:
```suggestion
      <Themed.h1 sx={{textAlign: "center", mb: 20}}>{title}</Themed.h1>
```

File: pages/fcl/authn.tsx [hunk 3]
```diff
@@ -48,8 +49,8 @@ function AuthnDialog({
 
   const onCancel = () => setEditingAccount(null)
 
-  if (!accounts && error) return <Err error={error} />
-  if (!accounts || isLoading) return null
+  if (error) return <Err error={error} />
```
Comment (huyndo) [inline pages/fcl/authn.tsx:52]:
```suggestion
  if (error) return <Err error={error} title="Authentication error"/>
```

File: contexts/AuthnContext.tsx [hunk 4]
```diff
@@ -36,7 +39,9 @@ export function AuthnContextProvider({children}: {children: React.ReactNode}) {
     initialize()
   }, [])
 
-  if (!isInitialized || !connectedAppConfig) return null
+  if (error) return <Err error={error} />
```
Comment (huyndo) [inline contexts/AuthnContext.tsx:42]:
```suggestion
  if (error) return <Err error={error} title="Initialization Error"/>
```

**Review by huyndo — CHANGES_REQUESTED** [review]
Thanks a lot @bartolomej for the PR! Everything looks good for me. I just got some suggestions before we merge:

1. Adding text wrap to the error message. I added the below to [styles/globals.css](https://github.com/onflow/fcl-dev-wallet/blob/a409ad0832bee2f224a52bfad23c2b898339dd5d/styles/globals.css) and it works pretty well. Lmk what you think. (Got it from [here](https://stackoverflow.com/questions/248011/how-do-i-wrap-text-in-a-pre-tag))
```
pre {
  white-space: pre-wrap;       /* Since CSS 2.1 */
  white-space: -moz-pre-wrap;  /* Mozilla, since 1999 */
  white-space: -pre-wrap;      /* Opera 4-6 */
  white-space: -o-pre-wrap;    /* Opera 7 */
  word-wrap: break-word;       /* Internet Explorer 5.5+ */
}
```
2. `title` param for `Err`. I added those as suggestions just to give more clarity on the error.
3. Removing `spinners-react` dependency. There's this code [snippet](https://gist.github.com/knowbody/578b35164b69e867ed4913423f6bed30) which gives a spinner using `styled-components` (which is already a dependency). You might encounter a certain error when trying it out, the solution can be found [here](https://stackoverflow.com/questions/41292559/could-not-find-a-declaration-file-for-module-module-name-path-to-module-nam).

Those are just my suggestions on the PR. Feel free to say what you think is suitable/not suitable. Thanks for contributing!

**Comment by bartolomej** [comment]
@huyndo Awesome! I think that makes sense! Added fixes for 2. and 3. suggestions.

For the text wrapping issue (1.) you mentioned that you already added this to the `globals.css`, but I can't see that in the [mentioned commit](https://github.com/onflow/fcl-dev-wallet/blob/a409ad0832bee2f224a52bfad23c2b898339dd5d/styles/globals.css). Should I add this on this branch?

**Review by sideninja — APPROVED** [review]

**Review by huyndo — APPROVED** [review]
Looks good to me @bartolomej. Thanks a lot!

**Comment by gregsantos** [comment]
Thanks for this great addition @bartolomej !

**Pull request merged**